src/circuit/circuit_frame.rs

```diff
@@ -2566,39 +2569,63 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
         let args_is_dummy =
             arg_t.alloc_equal(&mut cs.namespace(|| "args_is_dummy"), &g.dummy_arg_ptr)?;
 
-        let next_exp = AllocatedPtr::pick(
-            &mut cs.namespace(|| "pick nexp exp"),
+        let result_is_fun =
+            alloc_equal(cs.namespace(|| "result_is_fun"), function.tag(), &g.fun_tag)?;
+
+        let continuation_is_tail = alloc_equal(
+            &mut cs.namespace(|| "continuation is tail"),
+            continuation.tag(),
+            &g.tail_cont_tag,
+        )?;
+
+        let tail_cont = AllocatedContPtr::pick(
+            &mut cs.namespace(|| "the tail continuation"),
+            &continuation_is_tail,
+            &continuation,
+            &newer_cont,
+        );
+
+        let next_expr = AllocatedPtr::pick(
+            &mut cs.namespace(|| "pick nexp expr"),
```

Comment:
This misspelling of 'nexp' instead of 'next' seems to have been duplicated repeatedly (probably copy/paste).